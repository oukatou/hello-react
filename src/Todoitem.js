import React, { Component }  from 'react';

class Todoitem extends Component{
    delete(){
        const { handleDelete, index } = this.props;
        handleDelete(index)
    }
    render(){
        const {content} = this.props
        return (
            <li>{content}<span onClick={this.delete.bind(this)}>   Ã—</span></li>
        )
    }
}

export default Todoitem;