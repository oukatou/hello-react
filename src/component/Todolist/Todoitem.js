import React, { Component }  from 'react';

class Todoitem extends Component{
    delete(){
        const { handleDelete, index } = this.props;
        handleDelete(index)
    }
    render(){
        const {content,completed} = this.props
        return (
            <li style={{
                textDecoration: completed ? 'line-through' : 'none'
              }}>
              { content }
              <span onClick = {this.delete.bind(this)} 
                    style={{
                        display: completed ? 'none' : 'inline'
              }}>   Ã—</span> 
            </li>
        )
    }
}

export default Todoitem;