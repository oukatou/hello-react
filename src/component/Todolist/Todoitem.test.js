import React from 'react'
import {shallow} from 'enzyme'
import Todoitem from './Todoitem'
import renderer from 'react-test-renderer';

describe('<Todoitem/>', ()=>{

    let props={
        handleDelete: jest.fn(),
        content:'test1'
      }
    const todoitem = shallow(
        <Todoitem {...props}></Todoitem>
    )
    it('render todoitem',()=>{
        expect(todoitem.find('li').exists()).toEqual(true)
        expect(todoitem.find('li').text()).toEqual('test1   Ã— ')
        todoitem.find('span').simulate('click')
        expect(props.handleDelete).toBeCalled()
        let tree = renderer.create(<Todoitem {...props}></Todoitem>).toJSON()
        expect(tree).toMatchSnapshot()

    })
})