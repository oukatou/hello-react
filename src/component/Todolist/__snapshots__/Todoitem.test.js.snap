// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todoitem/> render todoitem 1`] = `
<li>
  test1
  <span
    onClick={[Function]}
  >
       Ã—
  </span>
   
</li>
`;
